subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -262,6 +235,33 @@ static int get_error_code_value(cJSON *const obj, enum nrf_cloud_error * const e
 	return 0;
 }
 
+static int info_encode(cJSON * const root_obj, const struct nrf_cloud_modem_info * const mdm_inf,
+	const struct nrf_cloud_svc_info * const svc_inf)
+{
+	int ret = 0;
+
+#ifdef CONFIG_MODEM_INFO
+	if (mdm_inf) {
+		ret = nrf_cloud_modem_info_json_encode(mdm_inf, root_obj);
+		if (ret) {
+			return -ENOMEM;
+		}
+	}
+#endif
+
+	if (svc_inf) {
+		cJSON *svc_inf_obj = cJSON_AddObjectToObjectCS(root_obj, JSON_KEY_SRVC_INFO);
```

Comment:
Could this object creation be moved into `nrf_cloud_service_info_json_encode`?

If you did that, then `info_encode` could be removed entirely, and replaced with just calls to `nrf_cloud_modem_info_json_encode` and `nrf_cloud_service_info_json_encode`

Reply:
Even if not splitting `info_encode`, doing this would increase `info_encode`'s symmetry

Reply:
i'll add this to IRIS-5446